### PR TITLE
[FIX] account_edi, web, mail: add attachment delete in public tour

### PR DESCRIPTION
**Pull request by seb-odoo**
This allows better coverage of the feature and in particular its access check.

Taken from https://github.com/odoo/odoo/pull/138330 to ease its diff.

### CODE REVIEW

File: addons/mail/static/tests/tours/discuss_channel_public_tour.js [hunk 1]
```diff
@@ -122,7 +122,13 @@ registry.category("web_tour.tours").add("discuss_channel_public_tour.js", {
...
             content: "Check edited message contains the extra attachment",
             trigger: '.o-mail-Message .o-mail-AttachmentCard:contains("extra.txt")',
-            run() {},
+            async run() {
+                await click(".o-mail-AttachmentCard-unlink", {
+                    parent: [".o-mail-AttachmentCard", { text: "extra.txt" }],
+                });
+                await click(".modal .btn.btn-primary");
+                await contains(".o-mail-AttachmentCard", { text: "extra.txt", count: 0 });
+            },
```
Comment (alexkuhn) [inline addons/mail/static/tests/tours/discuss_channel_public_tour.js:131]: Took me a bit of time to understand the "primary" button is to confirm a deletion. Would have been easier with a selector a bit more clearer

**Review by alexkuhn — APPROVED** [review]
@robodoo delegate+ r+

File: addons/mail/static/tests/tours/discuss_channel_public_tour.js (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/tests/tours/discuss_channel_public_tour.js:131]: The text is like "ok", I'm not sure if it's a lot more clear, and there are no specific classes to help that I saw.

**Comment by seb-odoo** [comment]
@robodoo r+

**Pull request closed without merging**